# 4.  Análisis por objetivos (Objetivo Específico 1)

Análisis Temporal (Objetivo Específico 1): Describir la evolución temporal de la cobertura, deserción, aprobación y repitencia en la educación preescolar, básica y media en los municipios de Colombia entre 2011 y 2023.

## 4.1 Configuración del Entorno e Importación de Módulos

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import sys
import os
  
# Añadir la carpeta 'src' y 'config' al path de Python,
sys.path.append(os.path.abspath(os.path.join('..', 'src'))),
sys.path.append(os.path.abspath(os.path.join('..', 'config'))),
   
# Importar funciones y variables desde tus módulos,
from data_loader import load_processed_data # Para cargar los datos ya limpios,
# Importar funciones de visualización,
from visualizer import(
    plotly_time_series,
    plotly_subcategories_time_series
) 
from utils import (
    temporal_statistics
)
from settings import PROCESSED_DATA_PATH, INDICADORES_ESTADISTICO, FIGURES_PATH,INDICADORES_PLOTLY, SUBCATEGORIAS_PLOTLY

## 4.2 Carga del Dataset Limpio

In [2]:
# Cargar el dataset limpio 
df_cleaned = load_processed_data(PROCESSED_DATA_PATH)

if df_cleaned is not None:
    print("✅ Datos limpios cargados exitosamente.\n")
    print(df_cleaned.head())
    print(f"Dimensiones del DataFrame limpio: {df_cleaned.shape}")
else:
    print("⛔ No se pudo cargar el DataFrame limpio. Asegúrate de haber ejecutado 01_exploracion_inicial.ipynb primero.")

Intentando cargar los datos procesados desde: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\data\processed\datos_educacion_limpios.csv
Datos procesados cargados exitosamente. Dimensiones: (14582, 34)
✅ Datos limpios cargados exitosamente.

   Unnamed: 0  anio  codigo_municipio   municipio  codigo_departamento  \
0           0  2023              5001    Medellín                    5   
1           1  2023              5002   Abejorral                    5   
2           2  2023              5004    Abriaquí                    5   
3           3  2023              5021  Alejandría                    5   
4           4  2023              5030       Amagá                    5   

  departamento  codigo_etc              etc  poblacion_5_16  \
0    Antioquia      3759.0         Medellín        377562.0   
1    Antioquia      3758.0  Antioquia (ETC)          3634.0   
2    Antioquia      3758.0  Antioquia (ETC)           503.0   
3    Antioquia      3758.0  Antioquia (E

### 4.3 Estadísticas básicas anuales

In [3]:
estadisticas_anuales = temporal_statistics(
    df_cleaned,
    INDICADORES_ESTADISTICO,
    grupo_temporal='anio' 
)

# Mostrar resultados
estadisticas_anuales.head(13)

✅ Estadísticas calculadas para 20/20 indicadores


anio  cobertura_neta_mean  cobertura_neta_median  cobertura_neta_std  \
0   2011            87.000624                 87.450           16.849926   
1   2012            85.585116                 86.400           17.025903   
2   2013            85.386453                 86.200           17.395132   
3   2014            86.777522                 87.425            4.929047   
4   2015            83.419492                 84.050           18.123789   
5   2016            83.543182                 82.750           22.651055   
6   2017            83.140722                 81.980           23.333339   
7   2018            88.121872                 89.170           14.657362   
8   2019            87.537638                 88.300           14.435617   
9   2020            86.371044                 86.980           14.558207   
10  2021            87.012756                 87.300           14.346948   
11  2022            85.622382                 85.750           15.235025   
12  2023            83.599197                 83.250           15.363757   

    cobertura_neta_primaria_mean  cobertura_neta_primaria_median  \
0                      86.970178                          87.900   
1                      85.079234                          86.250   
2                      83.932050                          84.450   
3                      77.387745                          78.480   
4                      81.019519                          82.225   
5                      81.217112                          80.460   
6                      80.043868                          78.450   
7                      85.608316                          86.660   
8                      84.547175                          85.155   
9                      82.932016                          83.460   
10                     83.218332                          83.210   
11                     82.152872                          82.250   
12                     80.088635                          80.130   

    cobertura_neta_primaria_std  cobertura_neta_secundaria_mean  \
0                     17.557144                       65.296533   
1                     17.810067                       65.625749   
2                     17.871325                       67.507736   
3                      8.449199                       68.365954   
4                     18.494704                       67.478004   
5                     22.778936                       68.637772   
6                     23.420520                       69.570508   
7                     14.567077                       73.149572   
8                     14.269902                       73.516551   
9                     14.200106                       73.822221   
10                    13.958339                       74.463595   
11                    14.648108                       72.508109   
12                    14.933771                       70.752676   

    cobertura_neta_secundaria_median  cobertura_neta_secundaria_std  ...  \
0                             67.100                      18.451098  ...   
1                             67.150                      18.928320  ...   
2                             68.900                      18.909304  ...   
3                             69.735                       9.088606  ...   
4                             68.810                      19.160355  ...   
5                             67.885                      22.183076  ...   
6                             69.085                      22.615400  ...   
7                             74.855                      17.901956  ...   
8                             74.900                      17.916682  ...   
9                             74.830                      17.390337  ...   
10                            74.960                      17.161749  ...   
11                            72.540                      17.402835  ...   
12                            71.080                      1

**1. Cobertura Neta**

| Aspecto       | Observación                                                                                                                                     |
|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
| *General*   | La cobertura neta es bastante alta en general, lo que indica que la mayoría de niños están matriculados en el grado que les corresponde por edad. |
| *Tendencia* | Hay una caída notoria en 2015, lo que es preocupante.                                                                                           |
| *Niveles*   | - **Primaria**: Muestra una baja fuerte en 2014 (77.4%), posiblemente por un cambio en la forma de medición o un evento puntual. <br> - **Media**: Tiene un aumento anómalo en 2014, lo que sugiere irregularidades o algún ajuste. |
| *Desigualdad* | La alta desviación estándar indica que la cobertura no es homogénea y varía mucho según el municipio.                                          |


**2. Deserción Escolar**

| Aspecto     | Observación|
|-------------|---------|
| *General* | La deserción tiende a disminuir entre 2011 y 2014, lo que es positivo, pero vuelve a aumentar levemente en 2015. |
| *Niveles* | Secundaria es el nivel con mayor deserción (4–5%), siendo el más vulnerable. |
| *Preocupación* | El repunte en 2015 puede estar relacionado con problemas en la calidad educativa o factores externos (económicos, sociales). |

**3. Aprobación Escolar**

| Aspecto        | Observación                                                                                                                                 |
|----------------|---------------------------------------------------------------------------------------------------------------------------------------------|
| *General*     | La aprobación es alta en general, con un pico en 2014, lo que muestra un buen desempeño en ese año.                                       |
| *Niveles*     | Secundaria tiene la tasa de aprobación más baja y además sufrió una caída en 2015, lo que coincide con el aumento en deserción.          |
| *Interpretación* | El descenso en la aprobación en 2015 puede ser un signo de deterioro en la calidad o en las condiciones de aprendizaje.                  |

**4. Repitencia Escolar**

| Aspecto     | Observación                                                                                                                                  |
|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|
| *General*       | La repitencia ha ido aumentando constantemente, lo que indica que más estudiantes están quedándose en el mismo grado.                         |
| *Niveles*       | Secundaria presenta el mayor aumento, con un pico en 2015.                                                                                     |
| *Preocupación*  | La repitencia alta suele ser un indicador negativo, ya que puede desmotivar a los estudiantes y está relacionada con la deserción futura.     |

**5. Reprobación Escolar**

| Aspecto | Observación |
|-------------|-----------------|
| *General* | La reprobación crece sobre todo en secundaria, alcanzando su nivel más alto en 2015. |
| *Niveles* | La tasa en secundaria es casi el doble que en primaria y media, lo que refleja la mayor dificultad en ese nivel. |
| *Impacto* | La alta reprobación explica en parte el aumento en la repitencia y, a largo plazo, podría aumentar la deserción. |

**Desviación estándar (std)**

- Los valores altos de std en indicadores como *cobertura, reprobación y repitencia* muestran grandes desigualdades entre municipios.
Esto quiere decir que en ciertos municipios la situación educativa es mucho más precaria que en otros, lo que indica un problema de inequidad territorial.

### 4.4  Visualización de evolución temporal (líneas)

4.4.1 Evolución promedio por indicador

In [4]:
plotly_time_series(
            df=estadisticas_anuales,
            time_col='anio',
            indicators=INDICADORES_PLOTLY,
            title='Evolución temporal de indicadores educativos promedio por año',
            yaxis_title='Valor Promedio',
            save_filename='evolucion_indicadores_plotly.html' # Se guardará en reports/figuras/
        )

✅ Figura Plotly guardada como HTML en: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\reports\figuras\evolucion_indicadores_plotly.html


El gráfico muestra la evolución temporal de indicadores educativos promedio entre 2011 y 2023 donde podemos destacar:

- **Estabilidad en cobertura y aprobación:** La cobertura neta y la aprobación han mantenido valores elevados y relativamente estables a lo largo de los años, lo que sugiere una continuidad en el acceso y éxito académico de los estudiantes.
- **Tendencias de deserción, repitencia y reprobación:** Estos indicadores presentan valores bajos y estables, lo que podría reflejar esfuerzos en políticas educativas para reducir el abandono escolar y mejorar la calidad de enseñanza.
- **Comparaciones de trayectoria:** La aprobación sigue una línea similar a la cobertura neta, indicando que los estudiantes que acceden a la educación tienden a completar su ciclo con éxito. Mientras tanto, la deserción, repitencia y reprobación mantienen valores bajos sin grandes fluctuaciones.
- **Evaluación de impacto en educación:** Esta estabilidad en indicadores clave podría mostrar que las condiciones del sistema educativo han sido constantes en la última década, aunque sería interesante analizar posibles influencias como cambios en políticas gubernamentales, pandemias o reformas educativas.

In [6]:
plotly_subcategories_time_series(
    df_cleaned,
    subcategories_dict=SUBCATEGORIAS_PLOTLY,
    time_col='anio',
    save_path_prefix=os.path.join(FIGURES_PATH, 'test_evolucion_nivel_')
)


 Generando gráficos de evolución temporal por subcategorías (5 indicadores principales)...


✅ Figura Plotly guardada como HTML en: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\reports\figuras\test_evolucion_nivel_cobertura_neta.html


✅ Figura Plotly guardada como HTML en: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\reports\figuras\test_evolucion_nivel_desercion.html


✅ Figura Plotly guardada como HTML en: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\reports\figuras\test_evolucion_nivel_aprobacion.html


✅ Figura Plotly guardada como HTML en: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\reports\figuras\test_evolucion_nivel_repitencia.html


✅ Figura Plotly guardada como HTML en: c:\Users\LENOVO\OneDrive\Documentos\python class\Dathon_Academic_MEN\reports\figuras\test_evolucion_nivel_reprobacion.html


En términos generales, la interpretación de las gráficas revela lo siguiente sobre la evolución del sistema educativo entre 2011 y 2023:
- Cobertura: La educación primaria mantiene una cobertura muy alta y estable. La secundaria ha experimentado un crecimiento constante en su cobertura, mientras que la media (bachillerato) también muestra una tendencia ascendente, aunque con un nivel de cobertura aún inferior a los otros dos.
- Deserción: La tasa de abandono escolar es consistentemente baja en primaria. Sin embargo, es notablemente más alta y volátil en secundaria y media, lo que indica un desafío significativo en la retención de estudiantes en estos niveles, con picos preocupantes alrededor de 2020-2021.
- Aprobación: La primaria registra tasas de aprobación excepcionalmente altas. La secundaria muestra una tendencia a la baja en sus tasas de aprobación en los últimos años, lo cual es motivo de preocupación. La media presenta tasas de aprobación intermedias y más estables que secundaria, aunque con un ligero descenso reciente.
- Rendimiento (Repitencia y Reprobación):
La repitencia (repitición de año) ha experimentado un aumento alarmante en todos los niveles a partir de 2020, siendo particularmente pronunciado en secundaria y primaria. Esto sugiere que más estudiantes están teniendo que repetir el mismo grado.
La reprobación (pérdida de asignaturas o año) también muestra una tendencia creciente en secundaria, lo que indica dificultades en el aprendizaje. En primaria y media, aunque hubo picos de reprobación alrededor de 2020, parecen haber disminuido ligeramente en los años más recientes.

En síntesis, el panorama general es el siguiente:

El sistema educativo ha logrado avances importantes en la ampliación de la cobertura, especialmente en secundaria y media. Sin embargo, enfrenta desafíos crecientes en la calidad y eficiencia interna, evidenciados por el aumento de la deserción, repitencia y reprobación, particularmente en los niveles superiores (secundaria y media).
Un punto crítico es el impacto notable de la pandemia de COVID-19 a partir de 2020, que parece haber exacerbado o provocado un deterioro en los indicadores de rendimiento (aumento de repitencia y reprobación) y en la retención (aumento de deserción) en todos los niveles, aunque con una recuperación parcial en algunos aspectos hacia 2023. La educación secundaria emerge como el nivel con mayores retos en términos de retención y resultados académicos.
 